ScoutSuite/providers/aws/aws_facade.py

```diff
@@ -0,0 +1,34 @@
+import boto3
+from botocore.session import Session
+from collections import Counter
+
+# TODO: Handle authentication better. I don't even know how it currently works. I think connect_service is called somewhere.
```

Comment:
It's handled by the `authenticate` method of each provider (e.g. https://github.com/nccgroup/ScoutSuite/blob/master/ScoutSuite/providers/aws/provider.py#L52). For AWS we're passing the CLI params to Opinel.